### PR TITLE
Stream type fixed and channel information added for xdf file..

### PR DESCRIPTION
**Pull request by csavur**
It was setting the wrong stream type and not reading the channels info from the xdf.

### CODE REVIEW

**Comment by Yida-Lin** [comment]
Thanks @csavur ! Looks good to me, but seems there are some problems with CI.

Hi @cboulay  would appreciate it if you could please help take a look at the CI issue, thank you!

File: XdfStreamer/xdfstreamer.cpp [hunk 1]
```diff
@@ -180,6 +180,21 @@ lsl::stream_info XdfStreamer::initializeLslStreamsForXdfData(const int samplingR
...
+    std::vector<std::map<std::string, std::string> > channels = this->xdf->streams[this->stream_idx].info.channels;
+
+    lsl::xml_element chns = info.desc().append_child("channels");
+    for(const std::map<std::string, std::string> &item : channels) {
+
+        std::vector<std::pair<std::string, std::string>> ch_info;
+        for(auto &[key, val] : item) {
+            ch_info.push_back(std::make_pair(key, val));
+        }
+        chns.append_child("channel")
```
Comment (cboulay) [inline XdfStreamer/xdfstreamer.cpp]: I think you're assuming that each channel will have 2 elements, probably like "label" and "unit". But this isn't necessarily true. It's probably better to first make a channel object, then do a loop based on how many things were pushed into ch_info:

```
lsl::xml_element chn = chns.append_child("channel");
for (const auto& kv_pair: ch_info) {
    chn.append_child_value(kv_pair.first, kv_pair.second);
}
```

Of course, the optimal solution would be to have a full traversal of the xdf_stream.info with some generic + recursive functions. But I know that's a lot of work, and getting 80% of the way there for 10% of the work is a good tradeoff here.

**Comment by cboulay** [comment]
The PR wasn't responsible for the failing build.

After a few failed attempts (sorry for the spam), I updated the GH Actions script to use the latest liblsl and libxdf, also Ubuntu release now uses focal (20.04). This is now in release 1.15.0.

@csavur - You can rebase your PR on main then it should run the scripts again. Or if you add any commits (see my comment above) then that should do it too.

File: XdfStreamer/xdfstreamer.cpp (hunk 1 above)
Comment (csavur) [reply XdfStreamer/xdfstreamer.cpp]: @cboulay you are right the assumption may not be correct. I will fix and re-submitted again. I am also working ob the multi-stream version of it. I am testing now, hopefully will submit soon.

**Pull request closed without merging**